Improved unit-test line coverage and quality.

dhermes opened the pull request:
In particular
- Upgraded some unit test modules to `unittest2`.
- Got to 100% line coverage in `test_jwt`, `test_clientsecrets`,
  and `test_service_account` by using `self.assertRaises`
  instead of `self.fail()`
- In some places used `self.assertRaises` as a context manager so
  the thrown exception can be inspected / compared against. This
  is done instead of manually catching the exception and
  holding on to it (`unittest2` backports this feature from
  `unittest` that was not available in Python 2.6)
- Added license to `test_multistore_file`
- Removed legacy `python2.4` hashbang from `test_jwt` and
  `test_service_account`
- Added `unittest2` as a test dependency in `tox.ini`
- Updated `test_clientsecrets.test_load_by_filename_missing_file`
  to use the `self.assertRaises` context manager (this was
  a problem in #349 and #350)
- Updated `test_jwt` module docstring to reflect actual purpose

dhermes commented:
@nathanielmanistaatgoogle PTAL

nathanielmanistaatgoogle commented on tests/test_jwt.py:
```diff
@@ -1,6 +1,4 @@
-#!/usr/bin/python2.4
-#
-# Copyright 2014 Google Inc. All rights reserved.
+# Copyright 2015 Google Inc. All rights reserved.
```
Don't "update" copyright years in license headers. New files should always get the year of their introduction, but existing license headers should not be altered.

dhermes replied:
Ahh I see. I only did it because I was there. Will revert (it'll make the diff look nicer)

nathanielmanistaatgoogle commented on tests/test_clientsecrets.py:
```diff
@@ -210,32 +210,33 @@ def test_validate_error(self):
-            except clientsecrets.InvalidClientSecretsError as e:
-                self.assertTrue(str(e).startswith(match))
+
+            first_caught_exception = exc_manager.exception
```
No need for a local field; use `exc_manager.exception` everywhere you wish now that you have it available for use.

(And analogous other places too.)

dhermes replied:
SGTM

dhermes committed:
Improved unit-test line coverage and quality.

In particular

- Upgraded some unit test modules to `unittest2`.
- Got to 100% line coverage in `test_jwt`, `test_clientsecrets`,
  and `test_service_account` by using `self.assertRaises`
  instead of `self.fail()`
- In some places used `self.assertRaises` as a context manager so
  the thrown exception can be inspected / compared against. This
  is done instead of manually catching the exception and
  holding on to it (`unittest2` backports this feature from
  `unittest` that was not available in Python 2.6)
- Added license to `test_multistore_file`
- Removed legacy `python2.4` hashbang from `test_jwt` and
  `test_service_account`
- Added `unittest2` as a test dependency in `tox.ini`
- Updated `test_clientsecrets.test_load_by_filename_missing_file`
  to use the `self.assertRaises` context manager (this was
  a problem in #349 and #350)
- Updated `test_jwt` module docstring to reflect actual purpose

dhermes commented:
@nathanielmanistaatgoogle Changes addressed. PTAL

nathanielmanistaatgoogle commented:
Looks good; merge when tests pass for the edited commit.

dhermes commented:
Thanks!
